plox/plox/interpreter.py

```diff
@@ -0,0 +1,117 @@
+from plox.expressions import Expr, Literal, Grouping, Unary, Binary
+from plox.token_types import TokenType, Token
+from plox.errors import LoxErrors, LoxRuntimeError
+from functools import singledispatchmethod
+
+
+class Interpreter:
+    def interpret(self, expr: Expr):
+        try:
+            value = self._evaluate(expr)
+            print(self._stringify(value))
+        except Exception as e:
+            LoxErrors.runtime_error(e)
+
+    @singledispatchmethod
+    def _evaluate(self, expr: Expr):
+        raise NotImplementedError
+
+    @_evaluate.register
+    def _(self, expr: Binary):
+        left = self._evaluate(expr.left)
+        right = self._evaluate(expr.right)
+        match expr.operator.type:
+            case TokenType.AND:
+                return self._is_truthy(left) and self._is_truthy(right)
+            case TokenType.OR:
+                return self._is_truthy(left) or self._is_truthy(right)
+            case TokenType.BANG_EQUAL:
+                return not self._check_equal(left, right)
+            case TokenType.EQUAL_EQUAL:
+                return self._check_equal(left, right)
+            case TokenType.GREATER:
+                self._check_number_operands(expr.operator, left, right)
+                return float(left) > float(right)
```

Comment:
You can skip float conversion in cases like these. After a call to `_check_number_operands` we already know we are dealing with floats. If you're doing the conversion for Mypy's sake you can use `typing.cast`, i.e.

```python
from typing import cast

cast(float, left) > cast(float, right)
```